src/Core/Engine.php

```diff
@@ -56,13 +56,17 @@ public function start(Run $run): void
     {
         $this->configure($run);
 
+        $this->scheduler->setNamespace($run->namespace);
+
         $this->eventDispatcher->dispatch(
             new RunStarting($run),
             RunStarting::NAME,
         );
 
-        foreach ($run->startRequests as $request) {
-            $this->scheduleRequest($request);
+        if ($this->scheduler->empty()) {
```

Comment:
Can you explain why this change is necessary?

Reply:
I did this because with the Redis Scheduler, stopping and starting a run will continue on from where it left off.

So if the scheduler already has items in the queue, we should skip adding the initial requests onto the end of the scheduler.

Reply:
I see. Then this should also be covered by tests, however.